glint doesn't parse args with dashes correctly

### Summary
If you have a component arg that has a dash in the name, glint does not handle it correctly.

### Error
```hbs
// template file

<p>You passed in {{@example-arg}}</p>
```

```sh
$ yarn glint

app/components/example-component.hbs:2:20 - error TS2304: Cannot find name 'arg'.

2   You passed in {{@example-arg}}
                     ~~~~~~~~~~~

app/components/example-component.hbs:2:20 - error TS2339: Property 'example' does not exist on type '{ 'example-arg': string; }'.

2   You passed in {{@example-arg}}
                     ~~~~~~~~~~~
```

This [demo repro](https://github.com/nlfurniss/glint-repro#glint-repro) shows the issue, and contains one commit where a camel case arg works `ExampleArg` but a dasherized one does not

--- a/packages/transform/__tests__/template-to-typescript.test.ts
+++ b/packages/transform/__tests__/template-to-typescript.test.ts
@@ -484,11 +484,27 @@ describe('rewriteTemplate', () => {
           );
         });
 
+        test('chained `this` path with a spinal-case key', () => {
+          let template = '{{this.foo-bar}}';
+
+          expect(templateBody(template)).toMatchInlineSnapshot(
+            `"χ.emitValue(χ.resolveOrReturn(𝚪.this[\\"foo-bar\\"])({}));"`
+          );
+        });
+
         test('`@arg` path', () => {
           let template = '{{@foo}}';
 
           expect(templateBody(template)).toMatchInlineSnapshot(
             `"χ.emitValue(χ.resolveOrReturn(𝚪.args.foo)({}));"`
+          );
+        });
+
+        test('`@arg` path with a spinal-case name', () => {
+          let template = '{{@foo-bar}}';
+
+          expect(templateBody(template)).toMatchInlineSnapshot(
+            `"χ.emitValue(χ.resolveOrReturn(𝚪.args[\\"foo-bar\\"])({}));"`
           );
         });
 

--- a/packages/transform/src/template-to-typescript.ts
+++ b/packages/transform/src/template-to-typescript.ts
@@ -995,12 +995,9 @@ export function templateToTypescript(
         let thisStart = template.indexOf('this', start);
         emit.text('𝚪.');
         emit.identifier('this', thisStart);
-        if (parts.length) {
-          emit.text('.');
-        }
         start = template.indexOf('.', thisStart) + 1;
       } else if (kind === 'arg') {
-        emit.text('𝚪.args.');
+        emit.text('𝚪.args');
         start = template.indexOf('@', start) + 1;
       }
 
@@ -1014,7 +1011,7 @@ export function templateToTypescript(
       if (kind === 'free') {
         emitIdentifierReference(head, start);
       } else {
-        emit.identifier(head, start);
+        emitPropertyAccesss(head, { offset: start, optional: false });
       }
 
       start += head.length;

